app_dart/lib/src/model/luci/user_data.dart

```diff
@@ -0,0 +1,44 @@
+// Copyright 2024 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:convert';
+import 'dart:typed_data';
+
+/// Small utility methods to create a single place where we are encoding and
+/// decoding strings from buildbucket. This prevents us missing decoding in
+/// certain places or decoding incorrectly.
+class UserData {
+  static Map<String, dynamic> decodeUserDataBytes(List<int> encodedBytes) {
+    return decodeUserDataString(String.fromCharCodes(encodedBytes));
+  }
+
+  static Map<String, dynamic> decodeUserDataString(String encoded) {
```

Comment:
Let's add a test to cover the logic here.

Reply:
Adding...

Reply:
Added.